addon/components/frost-table.js

```diff
@@ -54,12 +81,34 @@ export default Component.extend({
     return columns.map((column, index) => Object.assign({index}, column))
   },
 
+  @readOnly
+  @computed()
+  _isSelectable () {
+    return !isNone(this.get('onSelectionChange'))
+  },
+
   // == Functions =============================================================
 
+  setShift (event) {
+    if (!this.isDestroyed) {
+      this.set('_isShiftDown', event.shiftKey)
+    }
+  },
+
   // == DOM Events ============================================================
 
   // == Lifecycle Hooks =======================================================
 
+  init () {
+    this._super(...arguments)
+    const itemKey = this.get('itemKey')
+    if (itemKey) {
+      this.set('_itemComparator', function (lhs, rhs) {
```

Comment:
Sure. I see they serve a different purpose. In this case,` itemKey` should be a required prop (It's an option in prop type now). Otherwise, if consumer invokes the component without `itemKey`, and then try to select an item. It apparent will fail since `_select` action in `frost-table` and `_isItemSelected ` cp in `frost-table-row` will try to run `_itemComparator ` but no such method exist.